src/Microsoft.AspNet.TestHost/WebSocketClient.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Security.Cryptography;
+using Microsoft.Framework.Internal;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Internal;
+
+namespace Microsoft.AspNet.TestHost
+{
+    public class WebSocketClient
+    {
+        private readonly Func<IFeatureCollection, Task> _next;
+        private readonly PathString _pathBase;
+
+        internal WebSocketClient([NotNull] Func<IFeatureCollection, Task> next, PathString pathBase)
+        {
+            _next = next;
+            
+            // PathString.StartsWithSegments that we use below requires the base path to not end in a slash.
+            if (pathBase.HasValue && pathBase.Value.EndsWith("/"))
+            {
+                pathBase = new PathString(pathBase.Value.Substring(0, pathBase.Value.Length - 1));
+            }
+            _pathBase = pathBase;
+
+            SubProtocols = new List<string>();
+        }
+
+        public IList<string> SubProtocols
+        {
+            get;
+            private set;
+        }
+
+        public Action<HttpRequestMessage> ConfigureRequest
+        {
+            get;
+            set;
+        }
+
+        public async Task<System.Net.WebSockets.WebSocket> ConnectAsync(Uri uri, CancellationToken cancellationToken)
+        {
+            // clientRequest
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, uri);
```

Comment:
This is useful for the ConfigureRequest callback. Should I keep this or drop the ConfigureRequest?